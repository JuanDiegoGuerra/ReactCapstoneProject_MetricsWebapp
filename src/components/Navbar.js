import React, { useState } from 'react';
import SpeechRecognition, { useSpeechRecognition } from 'react-speech-recognition';
import './Modal.css';
import { NavLink } from 'react-router-dom';
import { FaMicrophone } from 'react-icons/fa';
import { IoMdSettings } from 'react-icons/io';
import { IoChevronBack } from 'react-icons/io5';

const Navbar = () => {
  const {
    transcript,
    listening,
    resetTranscript,
    browserSupportsSpeechRecognition,
  } = useSpeechRecognition();

  const [modal, setModal] = useState(false);

  const toggleModal = () => {
    setModal(!modal);
  };

  if (modal) {
    document.body.classList.add('active-modal');
  } else {
    document.body.classList.remove('active-modal');
  }

  if (!browserSupportsSpeechRecognition) {
    return <span>Browser does not support speech recognition.</span>;
  }
/* eslint-disable */
/* disabled linters only for future feature (MICROPHONE) */
  return (
    <header className="flex justify-between p-6 items-center bg-blue-3 text-text-color">
      <NavLink className="lato text-text-color w-1/4" to="/">
        <div className="flex items-center gap-1">
          <IoChevronBack className="text-base" />
          <p>See all</p>
        </div>
      </NavLink>
      <h1 className="font-bold lato text-xl">React_Capstone-Crypto_Watcher</h1>
      <div className="flex gap-2 w-1/4 justify-end">
        <button onClick={toggleModal}>
          <FaMicrophone />
        </button>
        {modal && (
        <div className="modal">
          <div onClick={toggleModal} className="overlay" />
          <div className="modal-content">
            <div>
              <p className='mic'>
                Microphone üéô : 
                {listening ? ' On' : ' Off'}
              </p>
              <div className='mic-btns'>
              <button onClick={SpeechRecognition.startListening}>Start ‚è∫</button>
              <button onClick={SpeechRecognition.stopListening}>Stop ‚èπ</button>
              <button onClick={resetTranscript}>Reset üîÅ</button>
              </div>
              <p className='tt'>{transcript}</p>
              <button className="close-modal" onClick={toggleModal}>
                CLOSE
              </button>
            </div>
          </div>
        </div>
        )}

        <IoMdSettings />
      </div>
    </header>
  );
};

export default Navbar;
