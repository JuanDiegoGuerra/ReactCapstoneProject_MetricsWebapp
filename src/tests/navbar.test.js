import 'regenerator-runtime/runtime';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Store from '../redux/store';
import Navbar from '../components/Navbar';

jest.mock('axios');

describe('testing Navbar components', () => {
  it('test Navbar component', () => {
    const { container } = render(
      <BrowserRouter>
        <Provider store={Store}>
          <Navbar />
        </Provider>
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
