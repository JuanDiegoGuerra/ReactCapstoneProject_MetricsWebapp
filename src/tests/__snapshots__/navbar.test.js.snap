// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Navbar components test Navbar component 1`] = `
<div>
  <span>
    Browser does not support speech recognition.
  </span>
</div>
`;
